django_filters/filters.py

```diff
@@ -655,100 +658,248 @@ def filter(self, qs, lookup):
+      Django expressions.
+
+      This argument is DEPRECATED.  You should always use the ``params`` argument.
+      Using both ``fields`` and ``params`` arguments will result in ``AssertionError``.
 
     * ``field_labels`` is an optional argument that allows you to customize
-      the display label for the corresponding parameter. It accepts a mapping
-      of {field name: human readable label}. Keep in mind that the key is the
-      field name, and not the exposed parameter name.
+      display labels for ``fields``.  It accepts a mapping of
+      {model field name: human readable label}. Keep in mind that the key is the
+      model field name, and not the exposed param name.
 
-    Additionally, you can just provide your own ``choices`` if you require
-    explicit control over the exposed options. For example, when you might
-    want to disable descending sort options.
+      Similar to the ``fields`` argument, ``field_labels`` is DEPRECATED.
+      You should always use the ``params`` argument where a customized
+      display label can be put into ``label`` field of the corresponding
+      ordering descriptors.  Using both ``field_labels`` and ``params``
+      arguments will result in ``AssertionError``
+
+    * ``choices`` is an optional argument that allows for explicit control
+      over the exposed options.  For example, you can use this argument
+      to disable descending sort options.
 
     This filter is also CSV-based, and accepts multiple ordering params. The
     default select widget does not enable the use of this, but it is useful
     for APIs.
-
     """
     descending_fmt = _('%s (descending)')
 
     def __init__(self, *args, **kwargs):
-        """
-        ``fields`` may be either a mapping or an iterable.
-        ``field_labels`` must be a map of field names to display labels
-        """
-        fields = kwargs.pop('fields', {})
-        fields = self.normalize_fields(fields)
-        field_labels = kwargs.pop('field_labels', {})
+        if 'params' in kwargs:
+            assert 'fields' not in kwargs, "'params' and 'fields' cannot be passed simultaneously"
+            assert 'field_labels' not in kwargs, "'params' and 'field_labels' cannot be passed simultaneously"
+
+        if 'fields' in kwargs:
+            deprecate("`fields` argument of OrderingFilter constructor is deprecated in favor of `params`")
+        if 'field_labels' in kwargs:
+            deprecate("`field_labels` argument of OrderingFilter constructor is deprecated in favor of `params`")
+
+        if 'params' not in kwargs:
+            fields = kwargs.get('fields', {})
+            fields = self.normalize_fields(fields)
+            field_labels = kwargs.get('field_labels', {})
+            params = self.fields_to_params(fields, field_labels)
+        else:
+            params = self.normalize_params(kwargs['params'])
 
-        self.param_map = {v: k for k, v in fields.items()}
+        self._params = params
```

Comment:
Looking at the code around `_params` is going against the style. Just use `self.params`

Reply:
Indeed, it seems that in the existing code underscores are only utilized to name backing fields used in the implementation of get/set properties. Renamed in edda682.